### PR TITLE
Feature/hooks before actions

### PR DESCRIPTION
**Pull request by phorward**
Additionally to the **onAdded**, **onEdited**, **onDeleted** hooks, this pull request introduces further hooks called before the specific database operation is performed, using **onAdd**, **onEdit** and **onDelete**.
Furthermore **onViewed** has been renamed to **onView** because the hook is performed before the operation takes place.

### CODE REVIEW

File: prototypes/singleton.py [hunk 1]
```diff
@@ -286,26 +300,26 @@ def onEdited(self, skel):
...
 		The default is doing nothing.
 
-		:param skel: The Skeleton that is viewed.
+		:param skel: The Skeleton that is being viewed.
 		:type skel: :class:`server.skeleton.Skeleton`
 
 		.. seealso:: :func:`view`
 		"""
-		pass
+		logging.debug("onView")
```
Comment (sveneberth) [inline prototypes/singleton.py:322]: I think this will spam the logs :/

File: prototypes/list.py [hunk 2]
```diff
@@ -546,7 +578,20 @@ def onViewed(self, skel):
 
 		.. seealso:: :func:`view`
 		"""
-		pass
+		logging.debug("onView")
```
Comment (sveneberth) [inline prototypes/list.py:581]: I think this will spam the logs :/

File: prototypes/singleton.py (hunk 1 above)
Comment (sveneberth) [inline prototypes/singleton.py:322]: this as well

File: prototypes/tree.py [hunk 3]
```diff
@@ -754,7 +783,20 @@ def onViewed(self, skel):
 
 		.. seealso:: :func:`view`
 		"""
-		pass
+		logging.debug("onView")
```
Comment (sveneberth) [inline prototypes/tree.py:786]: this as well

File: prototypes/list.py (hunk 2 above)
Comment (sveneberth) [inline prototypes/list.py:581]: I think this will spam the logs :/

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Comment by tsteinruecken** [comment]
I don't mind. For view it's probably better to remove the logging; but add/edit/deletion isn't that common...

**Comment by phorward** [comment]
All loggings had been replaced by passes.

**Review by sveneberth — APPROVED** [review]

**Pull request merged**